rust/blockstore/src/arrow/block/types.rs

```diff
@@ -561,6 +561,17 @@ pub enum BlockLoadError {
     NoRecordBatches,
 }
 
+impl ChromaError for BlockLoadError {
+    fn code(&self) -> ErrorCodes {
+        match self {
+            BlockLoadError::IOError(_) => ErrorCodes::Internal,
```

Comment:
note to self and others: Almost all of our codebase is littered with ErrorCodes::Internal, maybe evaluate if they can convey more than that?